### PR TITLE
Add setting to make touchbar controls optional

### PR DESCRIPTION
**Pull request by MarcusNoble**
**Related issues**: https://github.com/Microsoft/vscode/issues/63865, https://github.com/Microsoft/vscode/issues/38333, https://github.com/NascHQ/nasc-vscode-mac-touchbar/issues/9

Adds a new setting that allows the user to selectively have controls not appear in the touchbar without having to completely disable it. 

Example settings to hide all debug controls but keep navigation:
```json
{
	"keyboard.touchbar.ignoreActions": [
		"workbench.action.debug.start",
		"workbench.action.debug.run",
		"workbench.action.debug.restart",
		"workbench.action.debug.stepOver",
		"workbench.action.debug.stepInto",
		"workbench.action.debug.stepOut",
		"workbench.action.debug.stop",
		"workbench.action.debug.continue",
		"workbench.action.debug.pause"
	]
}
```

This also allows people to use plugins that make use of the touchbar without having to also have the nav and debug controls (that take up much of the space).

## Screenshots

Default:
image

Debug controls disabled:
image

Debug controls disabled with [nasc-vscode-mac-touchbar](https://github.com/NascHQ/nasc-vscode-mac-touchbar) extension enabled:
image

### CODE REVIEW

File: src/vs/code/electron-main/window.ts [hunk 1]
```diff
@@ -979,6 +979,8 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 		// of the group directly prevents ugly flickering from happening
 		this.touchBarGroups.forEach((touchBarGroup, index) => {
 			const commands = groups[index];
+			// Clear the segments first to prevent icon or label remaining
+			touchBarGroup.segments = [];
```
Comment (MarcusNoble) [inline src/vs/code/electron-main/window.ts]: There seems to be a bug in electron where it doesn't correctly clear icons or labels when segments are replaced (See: https://github.com/electron/electron/issues/17322). This gets around it by clearing out the segments then replacing them. No noticeable flicker when tested on my machine.

Comment (MarcusNoble) [reply src/vs/code/electron-main/window.ts]: This issue has been fixed in Electron (https://github.com/electron/electron/pull/17336) but has yet to be released.

Comment (MarcusNoble) [reply src/vs/code/electron-main/window.ts]: This has now been released in https://github.com/electron/electron/releases/tag/v3.1.7

What is the process for updating dependencies of VSCode? Are there any guidelines that need to be followed?

**Comment by bpasero** [comment]
Thanks, merged with the settings name `keyboard.touchbar.ignored`

**Pull request merged**